libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/LPC17xx_LPC40x8/NetworkInterface.c

```diff
@@ -0,0 +1,806 @@
+ * In most cases this will just involve calling whichever initialise function is provided with
+ * the Ethernet MAC peripheral drivers - which will in turn ensure the MAC hardware is enabled
+ * and clocked, as well as configure the MAC peripheral's DMA descriptors.
+ *
+ * xNetworkInterfaceInitialise() does not take any parameters, returns pdPASS if the initialisation was successful,
+ * and returns pdFAIL if the initialisation fails.
+ */
+BaseType_t xNetworkInterfaceInitialise(void)
+{
+   BaseType_t xReturn = pdFAIL;
+   static bool RxTaskCreated = pdFALSE; /* Guard against creating more than one receive task */
+   static bool HwInitialised = pdFALSE;
+   BaseType_t LinkStatus = 0;
+   BaseType_t RxFilterControl = 0;
+   TickType_t xLastWakeTime = xTaskGetTickCount(); /* Initialise with the current time. */
+
+   if (HwInitialised == pdFALSE)
+   {
+#if ( iplpcINIT_PORT_PINS == 1 )
+      /* Start by setting up the pins for the desired MAC-PHY interface type. */
+      prvBoard_ENET_Init(configUSE_RMII);
+#endif // ( iplpcINIT_PORT_PINS == 1 )
+
+      /* Initialise the peripheral using MII or RMII according to configuration */
+      Chip_ENET_Init(LPC_ETHERNET, iplpcUSE_RMII);
+
+      /* Initialise the PHY clock for 2.5MHz and address */
+      Chip_ENET_SetupMII(LPC_ETHERNET,
+                         Chip_ENET_FindMIIDiv(LPC_ETHERNET, 2500000),
+                         ENET_PHYDEF_PHYADDR);
+
+      /* Initialise the external PHY */
+      if (xPhyConfigure() == pdPASS)
+      {
+         /* Save MAC address */
+         Chip_ENET_SetADDR(LPC_ETHERNET, ucMACAddress);
+
+         /* Clear all MAC address hash entries. */
+         LPC_ETHERNET->RXFILTER.HashFilterH = 0;
+         LPC_ETHERNET->RXFILTER.HashFilterL = 0;
+
+         /* Add the station MAC address to the filter hash table */
+         prvAddMACAddress(ucMACAddress);
+
+#if (ipconfigUSE_LLMNR == 1)
+         /* Add the LLMNR MAC address to the filter hash table */
+         prvAddMACAddress (xLLMNR_MACAddress);
+#endif /* ipconfigUSE_LLMNR == 1 */
+
+         /* Enable packet reception according to configuration requirements. */
+         // AWM - for testing, we'll accept anything. TODO sort out config #defines
+         RxFilterControl = ENET_RXFILTERCTRL_APE; // Enable "perfect frames" as standard
+         //        RxFilterControl |= ENET_RXFILTERCTRL_AUE;        // Enable unicast frames
+         //        RxFilterControl |= ENET_RXFILTERCTRL_ABE;        // Enable broadcast frames
+         //        RxFilterControl |= ENET_RXFILTERCTRL_AME;        // Enable multicast frames
+         /* If the hash filters are non-zero, we are planning to use them, so enable
+          * multi- and unicast frames */
+         if ((LPC_ETHERNET->RXFILTER.HashFilterH != 0) || (LPC_ETHERNET->RXFILTER.HashFilterL != 0))
+         {
+            RxFilterControl |= ENET_RXFILTERCTRL_AMHE | ENET_RXFILTERCTRL_AUHE | ENET_RXFILTERCTRL_ABE;
```

Comment:
This doesn't agree with the comments on line 615/616, it is enabling broadcast frame reception as well with `ENET_RXFILTERCTRL_ABE`.